packages/opentelemetry-cloud-monitoring-exporter/test/monitoring.test.ts

```diff
@@ -135,6 +135,53 @@ describe('MetricExporter', () => {
       assert.strictEqual(result.code, ExportResultCode.FAILED);
     });
 
+    it('should return FAILED if project id promise is rejected', async () => {
+      await exporter['_projectId'];
+      exporter['_projectId'] = Promise.reject({
+        message: 'Failed to resolve projectId',
+      });
+
+      const result = await new Promise<ExportResult>(resolve => {
+        exporter.export([], result => {
+          resolve(result);
+        });
+      });
+
+      assert.deepStrictEqual(result, {
+        code: ExportResultCode.FAILED,
+        error: {
+          message: 'Await projectId failed: Failed to resolve projectId',
+        },
+      });
+    });
+
+    it('MetricsExporter#export should not raise an UnhandledPromiseRejectionEvent if projectId rejects', async () => {
+      const meter = new MeterProvider().getMeter('test-meter');
+      const labels: Labels = {['keya']: 'value1', ['keyb']: 'value2'};
+      const counter = meter.createCounter('name');
+      counter.add(10, labels);
+      await meter.collect();
+      const records = meter.getProcessor().checkPointSet();
+
+      await exporter['_projectId'];
+      exporter['_projectId'] = Promise.reject({
+        message: 'Failed to resolve projectId',
+      });
+
+      let unhandledPromiseRejectionEvent = false;
+      process.on('unhandledRejection', error => {
+        unhandledPromiseRejectionEvent = true;
+      });
+
+      const result = await new Promise<ExportResult>(resolve => {
+        exporter.export(records, result => {
+          resolve(result);
+        });
+      });
```

Comment:
Since we are still returning the promise, this would be equivalent?

```suggestion
      const result = await exporter.export(records, () => {});
```

Reply:
Oh yeah this makes sense. I was just following previous test patterns.

Reply:
gotcha. either way is fine with me, feel free to resolve

Reply:
Sounds good 👍🏻 Apologies for the major delay here. May has been quite a month for me. I'll get to wrapping this up by EoW.